crates/router/src/core/payments/helpers.rs

```diff
@@ -1583,7 +1584,7 @@ pub async fn retrieve_card_with_permanent_token(
         card_issuing_country: None,
         bank_code: None,
     };
-
+    cards::update_last_used_at(token, state).await?;
```

Comment:
We cannot use the `token` to update the payment method in the DB since it's a temporary identifier and is not stored in the table.

Reply:
Its the `pm_id` and not the temporary token, if you see we are updating the PaymentMethod using the pm_id